src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -6520,17 +6531,24 @@ void StorageReplicatedMergeTree::clearLockedBlockNumbersInPartition(
 
 void StorageReplicatedMergeTree::getClearBlocksInPartitionOps(
     Coordination::Requests & ops, zkutil::ZooKeeper & zookeeper, const String & partition_id, Int64 min_block_num, Int64 max_block_num)
+{
+    getClearBlocksInPartitionOpsImpl(ops, zookeeper, partition_id, min_block_num, max_block_num, "blocks");
+    getClearBlocksInPartitionOpsImpl(ops, zookeeper, partition_id, min_block_num, max_block_num, "async_blocks");
```

Comment:
This breaks any operation with partitions created on replicas before this code is merged (22.8 for example) because it tries to drop a non-existing key

Reply:
actually fails on the check inside the getClearBlocksInPartitionOpsImpl function not when it's dropped

Reply:
More in detail, the problem is that this function is looking for "async_blocks" in Zookeeper (in line 6543), which is created when the table is created. If the table was created by a replica with a previous version it is not there and throws an exception.
I have run a quick test checking if the "async_blocks" exists and in case it doesn't, adding it in the startup function, and it fixes the problem, but I don't know if it's the best solution.

Reply:
good investigation. I think it's good to check this new path during intialization (no matter restart or create table).

Reply:
Yep, it should be added to `createNewZooKeeperNodes` as well

Reply:
got it, will fix